matcher.go

```diff
@@ -30,6 +30,22 @@ func (matcher *EqMatcher) String() string {
 	return fmt.Sprintf("Eq(%v)", matcher.Value)
 }
 
+type NotEqMatcher struct {
+	Value Param
+}
+
+func (matcher *NotEqMatcher) Matches(param Param) bool {
+	return !reflect.DeepEqual(matcher.Value, param)
+}
+
+func (matcher *NotEqMatcher) FailureMessage() string {
+	return fmt.Sprintf("Expected: not %v; but got this value", matcher.Value)
```

Comment:
Not sure about this message, but I couldn't come up with anything that aligned with the messages for the existing matchers in a better way.

Reply:
How about:
```go
fmt.Sprintf("Expected: not %v; but got: %v", matcher.Value, matcher.Value)
```
It's a bit mechanical, but stays consistent with the other matchers, and maybe helps people to really see "aha, expected: not "abc", but got "abc".